azure-devops/asan-pipeline.yml

```diff
@@ -4,36 +4,46 @@
 # Build STL targeting x86 and x64, and run extra ASan testing
 
 variables:
-  tmpDir: 'D:\Temp'
-  buildOutputLocation: 'D:\build'
-  # Restrict to "stlasan" test
-  testSelection: '-R stlasan'
-
-pool:
-  name: 'StlBuild-2023-09-14T1251-Pool'
-  demands: EnableSpotVM -equals true
+  - template: config.yml
+  - name: testSelection
+    value: '-R stlasan' # Restrict to "stlasan" test
 
 pr: none
+trigger: none
+
+schedules:
+- cron: '0 2 * * *'
+  displayName: ASan-Daily-CI
+  branches:
+    include:
+    - main
 
 stages:
   - stage: Build_And_Test_x64
-    displayName: 'Build and Test'
+    displayName: 'Build and Test x64'
+    pool:
+      name: ${{ variables.poolName }}
+      demands: ${{ variables.poolDemands }}

```

Comment:
Yes, the `pool` info must be repeated in each stage like this: ["Expressions are only expanded for stages, jobs, steps, and containers (inside resources)."](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/template-expressions?view=azure-devops).